doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-ESP8266/index.md

```diff
@@ -5,4 +5,383 @@ permalink: /:path/
+
+</div><div id="Run" class="tab-pane fade" markdown="1" >
+
+#### Run mode
+
+| ESP8266-01 | USB-to-TTL |
+|:----------:|:----------:|
+| URXD       | TXD        |
+| UTXD       | RXD        |
+| CH_PD      | 3.3V       |
+| GND        | GND        |
+| VCC        | 3.3V       |
+
+</div></div>
+
+### NodeMCU
+
+Connecting NodeMCU is much simpler -- just connect it via micro-USB cable.
+
+## Installing requirements
+
+Prior to developing Kaa applications for ESP8266 some requirements should be installed.
+The detailed installation instructions can be found below.
+
+1. Prerequisites
+
+        sudo apt-get install autoconf libtool libtool-bin bison build-essential gawk git gperf flex texinfo libtool libncurses5-dev libc6-dev-amd64 python-serial libexpat-dev python-setuptools
+
+2. Set the `ESPRESSIF_HOME` variable.
+This variable will be used throughout installation process, and denotes a directory where ESP8266 SDK and toolchain will be placed.
+You are free to set it to whatever you like.
+
+            export ESPRESSIF_HOME=/opt/Espressif/
+
+3. Install toolchain
+
+        cd $ESPRESSIF_HOME
+        git clone -b lx106 git://github.com/jcmvbkbc/crosstool-NG.git
+        cd crosstool-NG
+        ./bootstrap && ./configure --prefix=`pwd` && make && sudo make install
+        ./ct-ng xtensa-lx106-elf
+        ./ct-ng build
+
+4. Add path to toolchain binaries to your .bashrc:
+
+        echo "export PATH=\$PATH:$ESPRESSIF_HOME/crosstool-NG/builds/xtensa-lx106-elf/bin" >> ~/.bashrc
+
+4. Install ESP8266 RTOS SDK
+
+        cd $ESPRESSIF_HOME
+        export ESP_SDK_HOME=$ESPRESSIF_HOME/esp-rtos-sdk
+        git clone https://github.com/espressif/esp_iot_rtos_sdk.git $ESP_SDK_HOME
+        cd $ESP_SDK_HOME
+        git reset --hard 169a436ce10155015d056eab80345447bfdfade5
+        wget -O lib/libhal.a https://github.com/esp8266/esp8266-wiki/raw/master/libs/libhal.a
+        cd $ESP_SDK_HOME/include/lwip/arch
+        sed -i "s/#include \"c_types.h\"/\/\/#include \"c_types.h\"/" cc.h
+
+5. Install esptool.py
+
+        cd $ESPRESSIF_HOME
+        git clone https://github.com/RostakaGmfun/esptool.git
+        cd esptool
+        sudo python setup.py install
+
+## Writing applications
+
+After successfull installation of requirements you can proceed to application development.
+This section shows how to setup minimal Kaa application for ESP8266.
+
+### Directory structure
+
+For this basic demo, the following directory structure is used:
+
+~~~
+CMakeLists.txt
+driver/
+    uart.h
+    uart.c
+ld/
+    eagle.app.v6.ld
+    eagle.rom.addr.v6.ld
+libs/kaa
+    <put Kaa SDK here>
+user/
+    user_main.c
+src/
+    kaa_demo.c
+~~~
+
+Some notes:
+
+* `CMakeLists.txt` is a CMake script (see below)
+* `driver/uart.c` and `driver/uart.h` files implement driver for ESP8266 UART interface.
+* The `ld` directory contains two linke scipts required for ESP8266 applications.
+* You should put generated Kaa C SDK tarball into `libs/kaa` directory and unpack it:
```

Comment:
You shouldn't put it in this director, but just unpack. We also have auto-unpack in some demos.

cc @GamovCoder